File: shuttle_client/src/types/invoice.ts
export interface InvoiceListItem {
  id: string;
  invoiceNumber: string;
  clientId: string;
  clientName: string;
  status: 'Draft' | 'Sent' | 'Paid' | 'Overdue' | 'Void';
  issuedDate: string;
  dueDate: string;
  totalAmount: number;
}

export interface InvoiceLineItem {
  id: string;
  tripId: string | null;
  lineType: 'PassengerService' | 'Cargo';
  description: string;
  unitRate: number;
  quantity: number;
  lineTotal: number;
  sortOrder: number;
}

export interface InvoiceDetail extends InvoiceListItem {
  notes: string | null;
  paidAt: string | null;
  subTotal: number;
  lineItems: InvoiceLineItem[];
}

File: shuttle_tablet/lib/features/clients/domain/entities/invoice.dart
import 'invoice_line_item.dart';

class Invoice {
  final String id;
  final String invoiceNumber;
  final String clientId;
  final String clientName;
  final String status; // Draft | Sent | Paid | Overdue | Void
  final DateTime issuedDate;
  final DateTime dueDate;
  final String? notes;
  final DateTime? paidAt;
  final double subTotal;
  final double totalAmount;
  final List<InvoiceLineItem> lineItems;

  const Invoice({
    required this.id,
    required this.invoiceNumber,
    required this.clientId,
    required this.clientName,
    required this.status,
    required this.issuedDate,
    required this.dueDate,
    this.notes,
    this.paidAt,
    required this.subTotal,
    required this.totalAmount,
    required this.lineItems,
  });
}
